Add plugin: Draw Steel Retainer

MartinDampier opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/MartinDampier/draw-steel-retainer/tree/1.0.4

## Release Checklist
- [ ] I have tested the plugin on
  - [X]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["description": "This plugin creates a initiative tracker that can be used for the MCDM rpg Draw Steel.",](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/manifest.json#L6)
Avoid including sentences like "This is a plugin that does ..." in your description, it should be a given since this is only shown when browsing plugins through the app or on our website.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

["authorUrl": "https://github.com/MartinDampier/draw-steel-retainer",](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/manifest.json#L8)
This should not point to the plugin repo, but your GH profile or website.

[table{](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/styles.css#L111-L122)
Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.

[// This adds a status bar item to the bottom of the app. Does not work on mobile apps.](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/main.ts#L24-L26)
Remove this sample code.

[console.log("Settings Loaded");](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/main.ts#L116), [console.log("Settings Saved");](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/main.ts#L121)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[.setButtonText("Add Player Character")](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/main.ts#L140), [.setName('Player Character')](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/main.ts#L158), [.setDesc('Set the PC\'s Name and Stamina')](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/main.ts#L159), [return 'Initiative Tracker';](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/Views/InitiativeTrackerView.ts#L42), [this.minionCountInput.selectEl.title = "Minion Count";](https://github.com/MartinDampier/draw-steel-retainer/blob/b05f9d9a3ba9322624cee5a016a970e99f14ce71/lib/Views/InitiativeTrackerView.ts#L84)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

MartinDampier commented:
Thank you for the feedback, I've gone ahead and implemented the suggestions.

- replaced the description with the new description: "Provides Director facing sidebar tools for the TTRPG Draw Steel"
- set author url to my github account
- removed these styles and converted them into classes that are used for the desired elements
- removed the sample code
- removed the logging
- changed titles in the settings tab to sentence case.

MartinDampier commented:
For documentation, released version 1.0.10 that includes all changes above and a missed spelling change.

claremacrae commented:
> # I am submitting a new Community Plugin
> ## Repo URL
> Link to my plugin: https://github.com/MartinDampier/draw-steel-retainer/tree/1.0.4

Here is the link to the actual repo:

https://github.com/MartinDampier/draw-steel-retainer/
